src/main/java/com/microsoft/azure/springcloudplayground/generator/ProjectGenerator.java

```diff
@@ -135,15 +136,31 @@ public File generateProjectStructure(ProjectRequest request) {
         }
     }
 
+    private void servicesPortsUpdate(@NonNull ProjectRequest request) {
+        Assert.isNull(request.getParent(), "should be parent project.");
+
+        final Set<Integer> set = new HashSet<>();
+        final Map<String, Integer> ports = request.getServicesPorts();
+        final List<String> services = request.getServices();
+
+        services.stream().filter(s -> ports.get(s) != null && ports.get(s) >= 1024 && ports.get(s) <= 65535)
+                .forEach(s -> set.add(ports.get(s)));
+
+        if (set.size() < services.size()) {
```

Comment:
does this mean duplicated ports existed?

Reply:
Yes, if not valid, will update all ports to default value.

Reply:
What if default value doesn't work somehow? Shall we do some validation on client side? @sophiaso

Reply:
The purpose is: let users provide the port, we do basic verification. If passed yet still break somehow, it isn't our responsibility then.

Reply:
So we only take care of fixed port (1-1023) for verification ?